src/components/other/ProjectAddForm/index.js

```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
-                    render={
-                        <SegmentInput
-                            options={projectVisibilityOptions}
-                            className={styles.isPrivateCheckbox}
-                            faramElementName="isPrivate"
-                            label={_ts('components.addProject', 'projectVisibilityInputLabel')}
-                            hint={_ts('components.addProject', 'projectVisibilityInputHint')}
-                        />
-                    }
-                />
-                <div className={styles.actionButtons}>
-                    <DangerButton onClick={this.onModalClose}>
-                        {_ts('components.addProject', 'modalCancel')}
-                    </DangerButton>
-                    <PrimaryButton
-                        type="submit"
-                        disabled={pending || !pristine}
-                    >
-                        {_ts('components.addProject', 'modalCreate')}
-                    </PrimaryButton>
-                </div>
-            </Faram>
-        );
-    }
+const projectAddSchema = {
+    fields: {
+        title: [requiredCondition],
+        isPrivate: [],
+        organizations: [],
+    },
+};
+
+function ProjectAddForm(props) {
+    const {
+        className,
+        onModalClose,
+        userGroups,
+        userId,
+        onProjectAdd,
+        setUserProject,
+        setUsergroupProject,
+        setUserProfileProject,
+    } = props;
+
+    const [faramValues, setFaramValues] = useState({ isPrivate: false, organizations: [] });
+    const [faramErrors, setFaramErrors] = useState({});
+    const [pristine, setPristine] = useState(true);
+
+    const [
+        pending,
+        ,
+        ,
+        createNewProject,
+    ] = useRequest({
+        url: 'server://projects/',
+        body: {
+            title: faramValues.title,
+            userGroups,
+            isPrivate: faramValues.isPrivate,
+            organizations: faramValues.organizations,
+        },
+        method: 'POST',
+        onSuccess: (response) => {
+            setUserProject({ project: response });
+            setUsergroupProject({ project: response });
+            setUserProfileProject({
+                userId,
+                project: response,
+            });
+            if (onProjectAdd) {
+                onProjectAdd(response.id);
+            }
+            notify.send({
+                title: _ts('components.addProject', 'userProjectCreate'),
+                type: notify.type.SUCCESS,
+                message: _ts('components.addProject', 'userProjectCreateSuccess'),
+                duration: notify.duration.MEDIUM,
+            });
+            onModalClose();
+        },
+        onFailure: (error, errorBody) => {
+            setFaramErrors(errorBody?.faramErrors);
+            setPristine(true);
+            notifyOnFailure(_ts('components.addProject', 'addProjectModalLabel'))({ error: errorBody });
+        },
```

Comment:
handle Fatal condition

Reply:
I was under the impression that the new request doesn't have fatal

Reply:
I've added a note